File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        unordered_map <int , int> mp;

        for(int num : nums){
            mp[num]++;
        }

        int presum = nums[0];
        int i = 1;

        while (i < nums.size()){
            if (nums[i] == nums[i-1]+1){
                presum += nums[i];
                i++;
            }else{
                break;
            }
        }

        while (mp.find(presum) != mp.end()){
            presum++;
        }

        return presum;

    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        mp = {}

        for num in nums:
            mp[num] = 1
        
        # print(mp)

        presum = nums[0]
        index = 1
        while index < len(nums):
            if nums[index] == nums[index-1]+1:
                presum += nums[index]
                index+=1
            else:
                break
        
        while presum in mp:
            presum += 1
        
        return presum

